### PR TITLE
DT-76: added enable/disable user for group manager if user is in grou…

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**
…p they are in

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
+import uk.gov.justice.digital.hmpps.oauth2server.maintain.AuthUserRoleService;
 import uk.gov.justice.digital.hmpps.oauth2server.nomis.model.StaffUserAccount;
 import uk.gov.justice.digital.hmpps.oauth2server.nomis.repository.StaffIdentifierRepository;
 import uk.gov.justice.digital.hmpps.oauth2server.nomis.repository.StaffUserAccountRepository;
 
 import javax.persistence.EntityNotFoundException;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.awt.geom.Ellipse2D;
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java]: Don't think that's a class we need!

Comment (Mjwillis) [reply src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java]: Please turn on optimise imports on all checked in code

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java [hunk 2]
```diff
@@ -71,21 +77,68 @@ public StaffUserAccount getUserByExternalIdentifier(final String idType, final S
...
+        for (GrantedAuthority authority : authorities) {
+            String grantedAuthorityAuthority = authority.getAuthority();
+            if ("ROLE_MAINTAIN_OAUTH_USERS".equals(grantedAuthorityAuthority)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void ensureUserLoggedInUserRelationship(final String loggedInUser, final Collection<? extends GrantedAuthority> authorities, final UserEmail userEmail) throws EnableDisableUserException {
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java]: Isn't this method already in another class?  Suggest make it a common.

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java (hunk 1 above)
Comment (SimonMitchellMOJ) [reply src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.java]: Agreed, will do ( thought I had turned on the optimise imports but must not have been on this project)

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserServiceTest.java [hunk 3]
```diff
@@ -77,49 +86,105 @@ public void findByEmailAndMasterIsTrue() {
...
+        final var user = new UserEmail("user");
+        final var group1 = new Group("group", "desc");
+        user.setGroups(Set.of(group1, new Group("group2", "desc")));
+
+        user.setAuthorities(new HashSet<>(List.of(new Authority("JOE", "bloggs"))));
+        final var groupManager = new UserEmail("groupManager");
+        groupManager.setGroups(Set.of(new Group("group3", "desc"), group1));
+        when(userEmailRepository.findByUsernameAndMasterIsTrue(anyString()))
+                .thenReturn(Optional.of(user))
+                .thenReturn(Optional.of(groupManager));
```
Comment (timharrison-moj) [inline src/test/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserServiceTest.java]: Multiple thenReturns? One for the subject user and one for the groupManager user in subsequent calls? Just checking its intended.

**Pull request merged**